app/Http/Controllers/Three/SignupsController.php

```diff
@@ -107,12 +110,14 @@ public function index(Request $request)
      */
     public function show(Request $request, Signup $signup)
     {
-        // Only allow an admin or the user who owns the signup to see the signup's unapproved posts.
-        if ($request->query('include') === 'posts') {
-            $signup = Signup::withVisiblePosts()->first();
+        // Only allow an admin, staff, or the user who owns the signup to see the signup.
+        $signup = Signup::whereVisible()->first();
```

Comment:
This is building a new database query from the `Signup` model & overwriting the instance we got from the router, so it'll give us the first post by the currently authenticated user. Instead, let's check against the instance we have:

```php
if (is_staff_user() || auth()->id() === $signup->northstar_id) {
  return $this->item($signup, 200, [], $this->transformer, $request->query('include'));
}
```